### ISSUE TITLE
Record data by SCAN4.5 and labrecorder is different?

### ISSUE THREAD

**Issue by wangxinzhi0** [labstreaminglayer/App-NeuroscanAcquire#1]
Hello,

        I'm going to save EEG data produced by Neuroscan Synamp II, but got some problems.

        First, data channels got one more. For example, sample*64channels got samples*65channels and the last channel number are all 65535. Is that a event channel?

        Second, record data by SCAN4.5 (.cnt) is different record data by labrecorder, likely difference by a multiple?

        Please give me a favor, thanks!

**Comment by cboulay** [comment]
The extra channel is an event channel:
https://github.com/labstreaminglayer/App-NeuroscanAcquire/blob/95f015990047385eb54b379605e7ecca2e783c49/nsatransporter.cpp#L72

As for the amplitude, you are probably correct that there is a scale factor missing.

The NeuroscanAcquire app simply reads the packet and passes it directly to LSL, either as an int32 or int16.

https://github.com/labstreaminglayer/App-NeuroscanAcquire/blob/95f015990047385eb54b379605e7ecca2e783c49/nsatransporter.cpp#L74-L83

You'll need to map that value to a float based on the resolution. This info is in the stream header:

https://github.com/labstreaminglayer/App-NeuroscanAcquire/blob/95f015990047385eb54b379605e7ecca2e783c49/nsatransporter.cpp#L60-L62

under `<settings><Resolusion>`. Note the unfortunate typo (it should be Resolution instead of Resolusion).

**Comment by wangxinzhi0** [comment]
Thanks for the quick reply! Is resolution a fixed float number if the hardware system doesn't change?

**Comment by cboulay** [comment]
I have no idea. I've never used this hardware. The number comes from an info packet from the device. The LSL app doesn't decide what the number is.

It's typical that the `uvolts / bit` is fixed for most EEG amplifiers, but this can sometimes change on some devices, especially if there are any "gain" settings.

It's easy enough to parse this from the stream info and use it to scale your data, so that's what I'd recommend doing.

**Comment by wangxinzhi0** [comment]
I got it. 
Is that mean I can get the float number by run **inlet.info.mResolution** [after L13](https://github.com/labstreaminglayer/liblsl-Matlab/blob/master/examples/ReceiveDataInChunks.m#L12-L13) in  **ReceiveDataInChunks.m**, 
and then use it to scale this chunk before saving it. 
Is it a feasible idea?

**Comment by cboulay** [comment]
Yes. I don't use Matlab much so I don't know the precise syntax, but something like that should work. You might have to use the specific xml functions. My first guess would be: `inlet.info().desc().child("settings").child_value("Resolusion");`

Note that will help you calculate the gain factor for real-time. If you want to apply the gain factor to the recorded data, then you should use the xdf importer.
